Update dependency vmware/govmomi to v0.20.3

misterikkit opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
/kind bug
/sig storage


**What this PR does / why we need it**:

Update dependency vmware/govmomi to v0.20.3
    
    
govmomi is the vSphere client library used by the vSphere cloud provider
and storage plugin. A bug in the SOAP client prevented storage classes
that use vSphere storage policies (aka SPBM) from working.
    
This bumps our dependency on vmware/govmomi from v0.20.1 to v0.20.3 to
pick up the fix in vmware/govmomi#1498
    
Here are all changes in the release:
https://github.com/vmware/govmomi/compare/v0.20.1...v0.20.3


**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes https://github.com/kubernetes/kubernetes/issues/79809

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

yastij commented:
/cc @dougm

dougm reviewed (commented):
@misterikkit I could also cherry-pick that fix to the govmomi release-0.20 branch and create a new tag.
I had created the branch to avoid kubernetes vendor churn such as this.  Any preference?

misterikkit commented:
> I could also cherry-pick that fix to the govmomi release-0.20 branch and create a new tag.

If that's not too much trouble, I think it would be easier to consume here. The cloud provider code is more or less in maintenance mode while it gets replaced with https://github.com/kubernetes/cloud-provider-vsphere .

dougm commented:
@misterikkit no problem at all, done: https://github.com/vmware/govmomi/compare/v0.20.2...v0.20.3

Commit:
Update dependency vmware/govmomi to v0.20.3

￼
misterikkit committed on Oct 4
govmomi is the vSphere client library used by the vSphere cloud provider
and storage plugin. A bug in the SOAP client prevented storage classes
that use vSphere storage policies (aka SPBM) from working.

This bumps our dependency on vmware/govmomi from v0.20.1 to v0.20.3 to
pick up the fix in vmware/govmomi#1498

Here are all changes in the release:
https://github.com/vmware/govmomi/compare/v0.20.1...v0.20.3

misterikkit commented:
Thanks very much @dougm! This PR now updates to v0.20.3.

liggitt commented:
lgtm, needs a rebase after https://github.com/kubernetes/kubernetes/pull/82809 merged

dougm reviewed (commented):
/lgtm

misterikkit commented:
Thanks for the prompt reviews. This has been rebased.

liggitt commented:
/lgtm
/approve

misterikkit commented:
@liggitt Does this belong in the v1.18 milestone, then?

liggitt commented:
sorry, was a race condition between a manual and bot-added label
/approve
